meshtastic/ble_interface.py

```
@@ -122,199 +153,259 @@ async def legacy_log_radio_handler(self, _, b):  # pylint: disable=C0116
+                for i in range(5):
+                    b = await self.client.read_gatt_char(FROMRADIO_UUID)
+                    if b:
+                        logger.debug(f"FROMRADIO read: {b.hex()}")
+                        self._handleFromRadio(b)
+                        return  # Exit after successful read
+                    logger.debug(
+                        f"No data received on attempt {
+                            i + 1}, retrying...")
+                    await asyncio.sleep(0.1)
+            except BleakError as e:
+                logger.error(f"Error reading from BLE: {e}")
```

Ruff (0.13.1)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)